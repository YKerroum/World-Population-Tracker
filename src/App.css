* {
  margin: 0;
  padding: 0;
}

.App {
  background-color: #5688e4;
  min-height: 100vh;
}
