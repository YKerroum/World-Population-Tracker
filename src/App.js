import React, { useEffect } from 'react';
import {
  Routes,
  Route,
} from 'react-router-dom';
import './App.css';
import { useDispatch } from 'react-redux';
import Countries from './pages/Countries';
import Home from './pages/Home';
import { getPopulation } from './redux/continents/homeSlice';

function App() {
  const dispatch = useDispatch();

  useEffect(() => {
  // get population of World '900' and continents 909,908....
    dispatch(getPopulation('900,909,908,935,904,903,905'));
  }, [dispatch]);

  return (
    <div className="App">
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/:idCity" element={<Countries />} />
      </Routes>
    </div>
  );
}

export default App;
