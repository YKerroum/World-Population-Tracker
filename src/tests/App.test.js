import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import App from '../App';

test('Application renders correctly', () => {
// Assign
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  // Assert
  expect(tree).toMatchSnapshot();
});
