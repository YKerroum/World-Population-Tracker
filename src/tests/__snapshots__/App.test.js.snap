// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application renders Link 1`] = `
<div
  className="App"
>
  <nav
    className="navbarStyle navbar navbar-expand-sd navbar-dark"
  >
    <div
      className="justify-content-between container-fluid"
    >
      <a
        aria-current="page"
        className="invisible active"
        href="/"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1.5em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 24 24"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M17.77 3.77L16 2 6 12l10 10 1.77-1.77L9.54 12z"
          />
        </svg>
      </a>
      <span
        className="navbar-text"
      >
        Continents
         
        Population List
      </span>
      <span
        className="navbar-text"
      >
        <svg
          fill="currentColor"
          height="1.5em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 16 16"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
          />
        </svg>
      </span>
    </div>
  </nav>
</div>
`;

exports[`Application renders correctly 1`] = `
<div
  className="App"
>
  <nav
    className="navbarStyle navbar navbar-expand-sd navbar-dark"
  >
    <div
      className="justify-content-between container-fluid"
    >
      <a
        aria-current="page"
        className="invisible active"
        href="/"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1.5em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 24 24"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M17.77 3.77L16 2 6 12l10 10 1.77-1.77L9.54 12z"
          />
        </svg>
      </a>
      <span
        className="navbar-text"
      >
        Continents
         
        Population List
      </span>
      <span
        className="navbar-text"
      >
        <svg
          fill="currentColor"
          height="1.5em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 16 16"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
          />
        </svg>
      </span>
    </div>
  </nav>
</div>
`;

exports[`renders Afric stats Link 1`] = `
<div
  className="App"
>
  <nav
    className="navbarStyle navbar navbar-expand-sd navbar-dark"
  >
    <div
      className="justify-content-between container-fluid"
    >
      <a
        aria-current="page"
        className="invisible active"
        href="/"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1.5em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 24 24"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M17.77 3.77L16 2 6 12l10 10 1.77-1.77L9.54 12z"
          />
        </svg>
      </a>
      <span
        className="navbar-text"
      >
        Continents
         
        Population List
      </span>
      <span
        className="navbar-text"
      >
        <svg
          fill="currentColor"
          height="1.5em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 16 16"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
          />
        </svg>
      </span>
    </div>
  </nav>
</div>
`;
